Code/Tools/SceneAPI/SceneBuilder/SceneSystem.cpp

```diff
@@ -57,6 +65,19 @@ namespace AZ
                 m_unitSizeInMeters = rootTransform.ExtractScale().GetMaxElement();
             }
 
+            if (!useOldAssimpBehavior)
+            {
+                // AssImp SDK internally uses a Y-up coordinate system, so we need to adjust the coordinate system to match the O3DE coordinate system (Z-up).
+                AZ::Matrix4x4 adjustmatrix = AZ::Matrix4x4::CreateFromColumns(
+                    AZ::Vector4(1, 0, 0, 0),
+                    AZ::Vector4(0, 0, 1, 0),
+                    AZ::Vector4(0, -1, 0, 0),
+                    AZ::Vector4(0, 0, 0, 1));
```

Comment:
You could also use `AZ::Matrix4x4::CreateFromColumnMajorFloat16` so you won't need to create temp Vector4's
```suggestion
                AZ::Matrix4x4 adjustmatrix = AZ::Matrix4x4::CreateFromColumnMajorFloat16(
                {
                    1.0f,  0.0f,  0.0f,  0.0f,
                    0.0f,  0.0f,  1.0f,  0.0f,
                    0.0f,-1.0f, 0.0f, 0.0f,
                    0.0f, 0.0f, 0.0f, 1.0f
                });
```